### PR TITLE
Add MACAddress to config.yaml

### PR DESCRIPTION
**Pull request by jp-bennett**
Allows setting a virtual MAC Address in config.yaml

### CODE REVIEW

**Review by thebentern — APPROVED** [review]

**Comment by jp-bennett** [comment]
> Humans are generally not good at generating random numbers, I'm afraid this would higher the chance of `deadbeef` node number collisions even more.
> 
> We could probably also try getting an Ethernet MAC if there's no Bluetooth, right?

Honestly I was just copying from my local Ethernet device. We could encourage that, or maybe even automate it.

**Comment by GUVWAF** [comment]
Yeah, I think that would be nice. 
Or, even when there's no Bluetooth/Ethernet, generate a random one and write it to the config file (in case you do a "factory reset").

**Comment by jp-bennett** [comment]
> Yeah, I think that would be nice. Or, even when there's no Bluetooth/Ethernet, generate a random one and write it to the config file (in case you do a "factory reset").

OK, I'll see about adding another option, to specify the network device to pull the MAC from. Should we consider the blank mac address an error condition? I'm thinking maybe yes, to prevent the proliferation of bogus devices.

**Comment by GUVWAF** [comment]
Sounds good, and yes, I think that should generate an (easily interpretable) error.

File: src/platform/portduino/PortduinoGlue.cpp [hunk 1]
```diff
@@ -137,6 +163,13 @@ void portduinoSetup()
         }
     }
 
+    uint8_t dmac[6];
+    getMacAddr(dmac);
+    if (dmac[0] == 128 && dmac[1] == 0 && dmac[2] == 0 && dmac[3] == 0 && dmac[4] == 0 && dmac[5] == 1) {
+        std::cout << "*** Blank MAC Address not allowed! " << std::endl;
```
Comment (GUVWAF) [inline src/platform/portduino/PortduinoGlue.cpp:169]: What about: `"MAC address should be set in 'config.yaml'"`? Then it's easier to figure out how to fix it.

Comment (jp-bennett) [reply src/platform/portduino/PortduinoGlue.cpp:169]: Yes. Still a WIP. I'll finalize the error message once the new options are available.

**Comment by jp-bennett** [comment]
Actually, is there a good reason we have the getMacAddr() function in portduino at all? That's not a standard Arduino function. I'm very much inclined to just pull the whole thing into the portduinoGlue file. Makes it all way easier

**Comment by GUVWAF** [comment]
I think it would indeed by fine to remove it, it's indeed not an Arduino function.

**Pull request merged**